### PR TITLE
ENH: Allow disabling histogram (2D views) interactions for LocalThreshold

### PR DESCRIPTION
**Pull request by mauigna06**

### CODE REVIEW

File: SegmentEditorLocalThreshold/SegmentEditorLocalThresholdLib/SegmentEditorEffect.py [hunk 1]
```diff
@@ -563,3 +567,5 @@ def cropOrientedImage(sourceImageData, roiNode):
 LABEL_VALUE = 1
 SELECTED_ISLAND_VALUE = 2
 OUTSIDE_THRESHOLD_VALUE = 3
+
+ENABLE_SLICE_VIEW_INTERACTION_PARAMETER_NAME = "ENABLE_SLICE_VIEW_INTERACTION"
```
Comment (lassoan) [inline SegmentEditorLocalThreshold/SegmentEditorLocalThresholdLib/SegmentEditorEffect.py]: Please follow naming conventions
```suggestion
ENABLE_SLICE_VIEW_INTERACTION_PARAMETER_NAME = "SliceViewInteractionEnabled"
```

**Review by lassoan — COMMENTED** [review]
It looks OK, but to integrate we need to have a good justification for allowing disabling interactions.

In general, we don't want to save any hidden state variables that cannot be edited by the users on the GUI. However, if we never need this feature then it must not pollute the GUI (and the scene).

Can you describe why this is needed? I suspect that there could be better ways to achieve what you want without introducing this change.

**Comment by mauigna06** [comment]
> Can you describe why this is needed? I suspect that there could be better ways to achieve what you want without introducing this change.

We have a qSlicerMarkupsPlaceWidget to put one point over the 2D slices, its position is used for the localThreshold seed and minimum threshold value (and the maximum threshold value is set to the upper scalar range of the CT), we consider the histogram interaction undesired and incompatible with our current workflow :)

File: SegmentEditorLocalThreshold/SegmentEditorLocalThresholdLib/SegmentEditorEffect.py (hunk 1 above)
Comment (mauigna06) [reply SegmentEditorLocalThreshold/SegmentEditorLocalThresholdLib/SegmentEditorEffect.py]: Done

**Comment by mauigna06** [comment]
@lassoan could this be merged? I think I completed your requirements

Thanks a lot! :)

**Comment by lassoan** [comment]
This is no longer needed, as view interactions can be disabled in the base class, by setting `effect.enableViewInteractions=False` - see:

https://github.com/Slicer/Slicer/blob/dc97ab25ebc1c595fc74cac4cb6ad93dfc894d58/Modules/Loadable/Segmentations/EditorEffects/Python/SegmentEditorEffects/SegmentEditorThresholdEffect.py#L747

**Pull request closed without merging**